src/authentication/steps/SelectServer.jsx

```diff
@@ -21,81 +22,92 @@ export class SelectServer extends Component {
   }
 
   componentDidMount() {
-    this.serverInput.focus()
+    this.input.focus()
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState({
+    this.setState(state => ({
+      ...state,
       error: nextProps.externalError ? ERR_WRONG_ADDRESS : null,
       fetching: nextProps.fetching
-    })
+    }))
   }
 
   componentDidUpdate() {
     if (this.state.error) {
-      this.serverInput.focus()
-      this.serverInput.select()
+      this.input.focus()
+      this.input.select()
     }
   }
 
-  validateValue(e) {
-    this.setState({ value: e.target.value, error: null })
+  onChange(value) {
+    this.setState(state => ({ ...state, error: null }))
+    this.setState(state => ({ ...state, value: value.trim() }))
   }
 
   onSubmit = async e => {
     e.preventDefault()
-    if (!this.state.value) return
+    const value = this.state.value
+    let error
 
-    let url = this.state.value
+    if (this.hasAtSign(value)) {
+      error = ERR_EMAIL
+    }
+    if (this.hasMispelledCozy(value)) {
+      error = ERR_COSY
+    }
 
-    if (url.indexOf('@') > -1) {
-      this.setState({ error: ERR_EMAIL })
-      return
+    const url = this.getUrl(value)
+    if (
+      !(url instanceof URL) ||
+      (url.protocol === 'http:' && !__ALLOW_HTTP__)
+    ) {
+      error = ERR_WRONG_ADDRESS
+    }
+    if (error) {
+      this.setState(state => ({ ...state, error }))
+      logException(error, {
```

Comment:
I saw in Sentry docs that it's better to pass an Error instance instead of a string to `Raven.captureException` (which is used by `logException`): https://docs.sentry.io/clients/javascript/usage/#raven-js-reporting-errors

Reply:
Ok, je modifie ça. J'ai du merger les conflits à la main suite à ta modif sur le reporter.